### ISSUE TITLE
Please make sure entity classes are rebuilt after "Ebean Enhancement" toggle is enabled.

### ISSUE THREAD

**Issue by kreiger** [ebean-orm/ebean-idea-enhancer#31]
This is related to issue #30

After enabling the "Ebean Enhancement" toggle i had to manually force a rebuild of the entity classes before my run configuration would work.

Please make sure that non-enhanced classes are rebuilt after the "Ebean Enhancement" toggle is enabled.

**Comment by rbygrave** [comment]
So right now there is a second alternative plugin that uses `javaagent` rather than `enhancement on compile` .. and we can use both types of enhancement together.

My grand plan is that the plugin supports both types of enhancement (agent & compile) and that we would turn on / enable the java agent based enhancement by default.  That would mean that we don't need to trigger a rebuilt.

In fact, many people may end up not using (enabling) the compile based enhancement. To date the javaagent based enhancement has been really good.  We get faster compile (as there is no enhancement during compile) but it is slightly slower running tests (as javaagent enhances at runtime) but people won't probably notice that unless they are enhancing many hundreds of entity beans (which we do when running the Ebean tests - maybe 0.5s slower for that).

**Comment by kreiger** [comment]
Where can i find the plugin that uses `javaagent` instead? Thanks!

**Comment by kreiger** [comment]
Nvm, just found https://github.com/ebean-orm-tools/ebean-idea-enhancer2

Will you put this in the Intellij Plugins Marketplace any time soon, or is it just an experiment until the plugin supports both?

**Comment by rbygrave** [comment]
I have published that plugin to the  Intellij Plugins Marketplace.  

So this this new plugin will replace this one.  So I have decided to NOT include the compile based enhancement in the new plugin - it only does javaagent based enhancement as I think that is going to be better.

We can release this compile based plugin under a different id/name if people still want to use it for some reason in the future.

So I'll close this with the view we all migrate to the new plugin (where there is no requirement to force a rebuild).
